### PR TITLE
Implement functionality for collection tags

### LINKED ISSUE #285: Complete functionality for collection tags
Each daily log should have tags indicating which collection it belongs to.

## Detailed Description
Users should be able to:
1. Add a daily log to a collection and see a new collection tag indicating which collection the daily log was added to
2. Remove a daily from a collection and see the corresponding collection tag disappear

When a collection is removed the relevant tags should also be removed from each daily log.

## Context
Collections are part of the core functionality of a bullet journal. Collection tags allow simple visual indication of which collections a daily log belongs to, allowing users to organize their entries.

## Possible Implementation
Drop-down menu allowing users to select which collection to add a daily log to.

## Your Environment
* Browser Name and version:
* Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by will-chung**
Initial implementation of collection tags which achieves data persistence. Depends on `current_log` property inside database. Styling conflict with date on daily log, will create issue to fix this.

### CODE REVIEW

**Comment by NoahT** [comment]
@yuzilyu these next two days, what I can do is go through and integrate `Router.js` changes so we can remove `current_collection` and `current_log` fields from indexedDB. What do you think?

File: source/javascript/src/daily.js [hunk 1]
```diff
@@ -119,6 +122,46 @@ collapse.addEventListener('click', () => {
...
+
+    const transaction = db.transaction(['currentLogStore'], 'readwrite')
+    const objectStore = transaction.objectStore('currentLogStore')
+    objectStore.openCursor().onsuccess = function (event) {
+      const cursor = event.target.result
+
+      if (cursor) {
+        const json = cursor.value
+        const collections = json.properties.collections
+        const currentLog = json.current_log
```
Comment (NoahT) [inline source/javascript/src/daily.js:143]: @will-chung  `current_log` and `current_collection` are going to be refactored/replaced soon; just a heads-up in cause you want to help me with these changes over the next two days

**Review by NoahT — APPROVED** [review]
LGTM

File: source/javascript/src/daily.js (hunk 1 above)
Comment (yuzilyu) [reply source/javascript/src/daily.js:143]: I think it's a good idea

File: source/models/mock_data.json [hunk 2]
```diff
@@ -8,6 +8,8 @@
     "collections": [{
       "type": "array",
       "name": "Collection one",
+      "logs": [
```
Comment (yuzilyu) [inline source/models/mock_data.json:11]: What is this field for?

**Review by yuzilyu — APPROVED** [review]

File: source/javascript/src/daily.js (hunk 1 above)
Comment (will-chung) [reply source/javascript/src/daily.js:143]: yeah we should definitely have a session soon to update all routing

File: source/models/mock_data.json (hunk 2 above)
Comment (will-chung) [reply source/models/mock_data.json:11]: `logs` contains an array of timestamps indicating which logs belong to a collection. It's used for achieving data persistence once a user adds a tag to a collection. We also considered adding a `tags` field to each daily log but doing it this way is a little more efficient.

**Pull request merged**